easybuild/tools/modules.py

```diff
@@ -273,13 +284,14 @@ def check_module_function(self, allow_mismatch=False, regex=None):
     def set_mod_paths(self, mod_paths=None):
         """Set mod_paths, based on $MODULEPATH unless a list of module paths is specified."""
         # make sure we don't have the same path twice, using nub
-        if mod_paths is not None:
-            self.mod_paths = nub(mod_paths)
-            for mod_path in self.mod_paths:
-                self.prepend_module_path(mod_path)
-        else:
+        if mod_paths is None:
             # no paths specified, so grab list of (existing) module paths from $MODULEPATH
             self.mod_paths = [p for p in nub(curr_module_paths()) if os.path.exists(p)]
+        else:
+            for mod_path in nub(mod_paths):
+                self.prepend_module_path(mod_path, set_mod_paths=False)
+            self.mod_paths = nub(mod_paths)
```

Comment:
positive conditions (`is None`) are easier to comprehend (especially if there's also an else), we try to do this throughout the whole codebase